warden/x/gmp/types/msgs.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgSetParams{}
+
+func NewMsgSetParams(
+	gmpAddress string,
+	gmpChannel string,
+	gmpTimeout int64,
+	feeRecipient string,
+	govAddress string,
+) *MsgSetParams {
+	params := &Params{
+		GmpAddress:   gmpAddress,
+		GmpChannel:   gmpChannel,
+		GmpTimeout:   gmpTimeout,
+		FeeRecipient: feeRecipient,
+	}
+	return &MsgSetParams{
+		Params:    params,
+		Authority: govAddress,
+	}
+}
+
+// Type implements LegacyMsg interface
+func (msg MsgSetParams) Type() string { return sdk.MsgTypeURL(&msg) }
+
+// ValidateBasic Implements sdk.Msg
+func (msg MsgSetParams) ValidateBasic() error {
+	// TODO validate params
+	return nil
```

Comment:
since we're doing this we could validate the params struct:

```suggestion
	return msg.Params.Validate()
```